libra-dev/main.c

```diff
@@ -0,0 +1,65 @@
+#include <stdio.h>
+#include <stdint.h>
+#include "account_address.h"
+
+int hexchr2bin(const char hex, char *out)
+{
+	if (out == NULL)
+		return 0;
+
+	if (hex >= '0' && hex <= '9') {
+		*out = hex - '0';
+	} else if (hex >= 'A' && hex <= 'F') {
+		*out = hex - 'A' + 10;
+	} else if (hex >= 'a' && hex <= 'f') {
+		*out = hex - 'a' + 10;
+	} else {
+		return 0;
+	}
+
+	return 1;
+}
+
+size_t hexs2bin(const char *hex, unsigned char **out)
+{
+	size_t len;
+	char   b1;
+	char   b2;
+	size_t i;
+
+	if (hex == NULL || *hex == '\0' || out == NULL)
+		return 0;
+
+	len = strlen(hex);
+	if (len % 2 != 0)
+		return 0;
+	len /= 2;
+
+	*out = malloc(len);
+	memset(*out, 0 , len);
+	for (i=0; i<len; i++) {
+		if (!hexchr2bin(hex[i*2], &b1) || !hexchr2bin(hex[i*2+1], &b2)) {
+			return 0;
+		}
+		(*out)[i] = (b1 << 4) | b2;
+	}
+	return len;
+}
```

Comment:
This isn't really necessary, I would prefer if we didn't include this hextobin logic and instead just have an array with the contents already in binary form.

Reply:
Yeah this was just for testing!